### DISCUSSION TITLE
On Subscription Callback is not coming when mutation query executed.

### DISCUSSION THREAD

**Discussion by vikrantgunjal** [hasura/graphql-engine#7387] [Question]
Hello,
I am facing a issue while implementing subscription, the callback is not coming with newly added data when mutation is happend.
Note : I am implementing this into c# core

**NuGet Packages Installed ->
GraphQL
GraphQL.client
GraphQL.Client.Serializer.Newtonsoft
GraphQL.SystemReactive**

Code attached - 

using System;
using GraphQL;
using GraphQL.Types;
using GraphQL.Client.Abstractions;
using GraphQL.Client.Http;
using GraphQL.Client.Serializer.Newtonsoft;

namespace GraphQLSubs
{
    class Program
    {
        static void Main(string[] args)
        {
            try
            {
                var graphQLClient = new GraphQLHttpClient("https://--Link---.hasura.app/v1/graphql", new NewtonsoftJsonSerializer());
                graphQLClient.HttpClient.DefaultRequestHeaders.Add("x-hasura-admin-secret", "---secret----");

                var query = new GraphQLRequest
                {
                    Query = @"
                    subscription MySubscription {
                      Notification {
                        Message
                        NewGuid
                        Signaltype
                        Status
                        Updatedon
                        UserId
                      }
                    }"
                };

                IObservable<GraphQLResponse<NotificationDataGraphQL>> subscriptionStream = graphQLClient.CreateSubscriptionStream<NotificationDataGraphQL>(query);

                var subscription = subscriptionStream.Subscribe(response =>
                {
                    **//Callback suppose to come here when mutation happens**   **// <===============**
                    Console.WriteLine($"Notification '{response.Data.Notification.NewGuid}' Added");
                });

                Console.ReadKey();
            }
            catch (Exception e)
            {
                Console.WriteLine(e.Message);
            }
        }
    }

    public class NotificationDataGraphQL
    {
        public NotificationFieldsData Notification { get; set; }
        //public NotificationFieldsData notificationAdded { get; set; }
    }

    public class NotificationFieldsData
    {
        public string NewGuid { get; set; }
        public string Signaltype { get; set; }
        public string Message { get; set; }
        public string UserId { get; set; }
        public string Status { get; set; }
        public string Updatedon { get; set; }
    }
}

This code works perfectly fine with own graphql server, but not working in hasura graphql
Please help me with this where do i go wrong???

**Comment by praveenweb** [comment]
Are you able to see the response on the websocket connection whenever new data comes in? I would start debugging by looking at where the problem is, either the server doesn't return the response, in which case the query needs to be looked at. If server seems to return a response (can be checked via network calls), then it could be a client side issue.

On a side note, it is not recommended to expose `x-hasura-admin-secret` from a client that is being sent as a header.
